### ISSUE TITLE
$popup does not work

### ISSUE THREAD

**Issue by dimisa-RUAdList** [uBlockOrigin/uBlock-issues#1785]
### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### Description

$popup does not work

### A specific URL where the issue occurs

`https://trashbox.ru/link/2021-10-26-rawdraw`

### Steps to Reproduce

1. Add to "My Filters" `trashbox.ru#@#a[href*="://trashbox.ru/r.php?"]`
2. Open `https://trashbox.ru/link/2021-10-26-rawdraw`
3. Click on the banner - https://i.imgur.com/KGMNLFv.jpg

### Expected behavior

The popup should be blocked by the rule `||trashbox.ru/r.php?$popup`, preventing the click on the advertising link.

### Actual behavior

There is a transition to the advertising link.

### uBlock Origin version

uBlock Origin development build 1.38.7b9 (default + RUS: RU AdList)

### Browser name and version

Google Chrome 95.0.4638.54 

### Operating System and version

Windows 7

**Comment by gorhill** [comment]
It's difficult for me to make change to the current behavior without seeing actual, real filter issue cases to convince me that the actual behavior _must_ be changed -- the one you provide is artificial, it's not going to happen in the real world.

**Comment by dimisa-RUAdList** [comment]
Okay, do as you see fit. If real cases appear, it will be possible to return to this later.

**Comment by Yuki2718** [comment]
Is a popup filter still needed when cosmetic filter already have done the job?

**Comment by dimisa-RUAdList** [comment]
No, I don't need it. But the current heuristic will disable pop-up blocking in cases where the advertising link is explicitly specified. For example, when branding the background.

**Comment by Yuki2718** [comment]
Also transparent link to trick user is common, but they should be addressed first by either cosmetic filter or scriptlet, shouldn't they?

**Comment by dimisa-RUAdList** [comment]
Yes, exactly.

**Comment by uBlock-user** [comment]
We usually fix popups in uBO with `##+js(nowoif)`, does it work for this case ?

**Comment by gorhill** [comment]
> does it work for this case ?

You don't need to do anything for the case provided here, you have to disable a cosmetic filter and click on an ad for it to happen -- something nobody is going to do.

**Comment by uBlock-user** [comment]
> click on an ad for it to happen

If I have make an ad, which is hidden already, reappear and then click on it just to prove a point, seems not worth the time.

**Comment by Yuki2718** [comment]
I personally feel the difference btwn ABP & uBO here would hardly be a matter at least in terms of false negative. OTOH the ABP's way of $popup can really be a problem, a worse case of the click-through issue where average user has no chance to choose. Hmm,, I certainly saw a similar case actually at AG, can't remember soon.

**Comment by Yuki2718** [comment]
Maybe https://github.com/AdguardTeam/AdguardFilters/issues/90055 Already fixed by EL but IIRC uBO didn't suffer at the time.

**Comment by uBlock-user** [comment]
As for the differences in behaviour with `$popup` with regards to ABP and uBP, the only ideal solution for that would be for both the developers of the extension(ABP and uBO) to come together, write a spec for `$popup` and implement it, so the behaviour is common and there are no differences( like it was done for `$removeparam` with AG.)

**Comment by dimisa-RUAdList** [comment]
Yes, that would be the perfect solution.

**Comment by LucianAndries** [comment]
Hello.

I have a similar issue on a website. It always opens a new window with a specific address. It does it no matter where I click, it does it even by reloading a page................
The popup is not related to anything, because all the links are different from the popup. I have no idea what triggers it, it's not a redirect either.

This popup: http://aniwatcher.com/search?q=ready

And knowing the link, I created custom filters for it. Filters like:
||aniwatcher.com/search?q=ready$popup,important
||aniwatcher.com/search?q=ready$popunder,important
||aniwatcher.com/search?q=ready

And the thing still opens this popup. And don't get me started with this "about:blank"...............
And if nobody figured it out, this is the website "aniwatcher.com".

Are there any more special commands to block these stupid popups? :(

I keep uBlock on default settings, using the built-in filters. And there's no filter that interferes with mine.
Also, I'm not using any dynamic filtering, I know those break everything.
I'm using latest Chrome, and latest Windows 10 Pro.

So does anyone have more ideas or advices in how to force block it? Because if that's it, then it has to be a bug, right? :(
Also, I only want to block that link, not all popups on that domain. Because I tend to open new shows in new tabs. Otherwise I could just block everything.

**Comment by Yuki2718** [comment]
@LucianAndries I don't get any popup. Probably better to report with built-in reports with all the details.
